packages/relay/src/lib/eth.ts

```diff
@@ -517,13 +518,48 @@ export class EthImpl implements Eth {
     }
   }
 
+  /**
+   * Do the pre-checks for the sendRawTransaction route
+   *
+   * @param transaction
+   * @private
+   */
+  private async precheckSendRawTransaction(transaction: string) {
+    const tx = ethers.utils.parseTransaction(transaction);
+    const rsTx = await ethers.utils.resolveProperties({
+      gasPrice: tx.gasPrice,
+      gasLimit: tx.gasLimit,
+      value: tx.value,
+      nonce: tx.nonce,
+      data: tx.data,
+      chainId: tx.chainId,
+      to: tx.to
+    });
+    const raw = ethers.utils.serializeTransaction(rsTx);
+    const recoveredAddress = ethers.utils.recoverAddress(
+      ethers.utils.arrayify(ethers.utils.keccak256(raw)),
+      // @ts-ignore
+      ethers.utils.joinSignature({ 'r': tx.r, 's': tx.s, 'v': tx.v })
+    );
+    const accountInfo = await this.mirrorNodeClient.getAccount(recoveredAddress);
+
+    if (accountInfo && accountInfo['ethereum_nonce'] > tx.nonce) {
+      throw predefined.NONCE_TOO_LOW;
+    }
+
+    if (accountInfo && accountInfo['ethereum_nonce'] + 1 < tx.nonce) {
```

Comment:
I don't think we can reliably handle this in the relay.  Consider where a robot is spamming really fast.  They may want to queue up multiple transactions before the account info settles to the new value on the mirror node.  This will ruin the TPS if they need to wait for the TX to do a full transit into the ledger and then through the mirror node.

Reply:
Nonce too low is fine.  Number only ever goes up.  So once it's too low it's always too low.

Reply:
If we add a 2-4 second cache for "latest nonce used by account" do you think that it would be sufficient?

Reply:
That won't scale horizontally.  Plus we need to consider robots sending txes to other relay clusters, or directly to the consensus nodes.